### ISSUE TITLE
Training on images of raspberry bi captured some days

### ISSUE THREAD

**Issue by zeynali** [AlexeyAB/darknet#2611]
Hi @AlexeyAB,
I captured the images of cars from highways with raspberry'pi camera, I have some question.
I want to detect various types of cars, and some of the frames are busy, because of the density of cars, 
Q1- Is it necessary to have labels (boxes) for all of the cars? if some are cars in some frames don't have labels, is this be a problem for training? that means, is necessary to have labeled the cars?
Q2- I captured the images with 25 FPS, and maybe the some of cars same in some frames, Is cause the overfitting problem? assume 50'th frame be in training set and 51'th frame be validation set, it appears these frames are same, Is cause the overfitting problem?
Q3- Because the frames are high height, the cars aren't very cleary, Is necessary any image pre-processing techniques?

**Comment by AlexeyAB** [comment]
@zeynali Hi,

1. You must label every car: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

> check that **each object are mandatory labeled in your dataset** - no one object in your data set should not be without label. In the most training issues - there are wrong labels in your dataset (got labels by using some conversion script, marked with a third-party tool, ...). Always check your dataset by using: https://github.com/AlexeyAB/Yolo_mark

2. If many of your images are similar in the Training dataset then you must have large dataset to avoid overfitting, more than 2000 images per class.
If you have very simillar images in the Training and Validation dataset, then the mAP on Validation dataset will increase even after overfitting, so it can lead to overfitting too.

3. Can you show an example?

**Comment by zeynali** [comment]
[www.google.com](url)

**Comment by zeynali** [comment]
[mandatory](url)

**Comment by zeynali** [comment]
https://i.ibb.co/QPgtLV3/1.jpg

**Comment by AlexeyAB** [comment]
Is it part of image or whole image?

If it is a part of image - then this is a good object for neural network. It will be deceted well.

**Comment by zeynali** [comment]
That images are for binary classification, and these are cropped from whole image,

**Comment by AlexeyAB** [comment]
So what do you want to use Detector or Classifier?

You shouldn't do any preprocessing.

You should have ~200-2000 different images per class to avoid overfitting.
If every 10 images are simillar, then you should have 10x more images to avoid overfitting, so you should have 2000 - 20 000 images per class.

**Comment by zeynali** [comment]
I have ~5K images for every class, but i don't know why my test result of classification on online stream of images is unstable? that is , when I croped the same car from 5'th frame and the result of network is class 0 , then I crapped the same car from 6'th frame, the result of network be class 1, and sensitive on conditional of weather. while I gathered any time of days images.
